fuel-p2p/src/request_response/codec.rs

```diff
@@ -0,0 +1,110 @@
+use super::messages::{RequestMessage, ResponseMessage};
+use async_trait::async_trait;
+use futures::{AsyncRead, AsyncWriteExt};
+use libp2p::{
+    core::{
+        upgrade::{read_length_prefixed, write_length_prefixed},
+        ProtocolName,
+    },
+    request_response::RequestResponseCodec,
+};
+use std::io;
+
+pub const REQUEST_RESPONSE_PROTOCOL_ID: &[u8] = b"/fuel/req_res/0.0.1";
+/// Max Size in Bytes of the messages
+// todo: this will be defined once Request & Response Messages are clearly defined
+const MAX_REQUEST_SIZE: usize = std::mem::size_of::<RequestMessage>();
+const MAX_RESPONSE_SIZE: usize = std::mem::size_of::<ResponseMessage>();
+
+#[derive(Debug, Clone)]
+pub struct MessageExchangeProtocol;
+
+impl ProtocolName for MessageExchangeProtocol {
+    fn protocol_name(&self) -> &[u8] {
+        REQUEST_RESPONSE_PROTOCOL_ID
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct MessageExchangeCodec {}
```

Comment:
Maybe we should denote that this is the bincode codec? I'm assuming we'll eventually have multiple codecs.